### PR TITLE
Feat/posts only editable for ten minutes

### PR DESCRIPTION
**Pull request by Hives**

### CODE REVIEW

**Comment by Hives** [comment]
[Users can only edit their posts up to 10 minutes after they were posted [SMALL]](https://trello.com/c/vqfWQEOm/56-users-can-only-edit-their-posts-up-to-10-minutes-after-they-were-posted-small)

File: Gemfile [hunk 1]
```diff
@@ -38,17 +38,18 @@ group :development, :test do
...
-  gem "simplecov-console"
   # for linting.
   gem "rubocop"
   gem "rubocop-performance"
+  gem "selenium-webdriver"
+  # for test coverage
+  gem "simplecov"
+  gem "simplecov-console"
+  gem "timecop"
 end
```
Comment (arthurashman) [inline Gemfile]: Can we get these in alphabetical order at some point?

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -35,9 +35,14 @@ def create
   end
 
   def update
-    @post = Post.find(params[:id])
+    post = Post.find(params[:id])
+
+    unless post.isEditable
+      flash[:danger] = "Could not edit post. Posts are only editable for 10 minutes."
```
Comment (arthurashman) [inline app/controllers/posts_controller.rb:41]: Would it be at all possible to hide/grey out the edit button after 10 minutes?

Comment (j-dav) [reply app/controllers/posts_controller.rb:41]: See index.html.erb, we have put this in on lines 23 to 25 already
Code:
        <% if post.isEditable %>
          <p><%= link_to 'Edit', edit_post_path(post) %></p> 
        <% end %>

**Comment by Hives** [comment]
If you leave the page open for 10 minutes then the link won't disappear
though which I think is what Arthur is suggesting? But I think that's such
a niche issue it's not worth worrying about, certainly at this stage.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -35,9 +35,14 @@ def create
   end
 
   def update
-    @post = Post.find(params[:id])
+    post = Post.find(params[:id])
+
+    unless post.isEditable
```
Comment (Hives) [inline app/controllers/posts_controller.rb]: I just realised I gave these methods JS-style names. Should be snake_case.

Comment (Hives) [reply app/controllers/posts_controller.rb]: And Ruby convention for predicate methods is that it should be like `.editable?`

File: app/models/post.rb [hunk 4]
```diff
@@ -3,4 +3,14 @@
...
   belongs_to :user
   delegate :email, :to => :user
+  
+  def isEditable
+    isLessThan10MinutesOld
+  end
+
+  private
+
+  def isLessThan10MinutesOld
```
Comment (Hives) [inline app/models/post.rb]: Should be `.less_than_ten_minutes_old?`

File: app/models/post.rb [hunk 5]
```diff
@@ -3,4 +3,14 @@
 class Post < ApplicationRecord
   belongs_to :user
   delegate :email, :to => :user
+  
+  def isEditable
```
Comment (Hives) [inline app/models/post.rb]: Should be `.editable?`

**Comment by j-dav** [comment]
Oh I see... Yeah probably some jquery implementation or something for that!

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (j-dav) [reply app/controllers/posts_controller.rb]: Agreed. Go with editable?

File: app/models/post.rb (hunk 4 above)
Comment (j-dav) [reply app/models/post.rb]: Goddam language conventions

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (Hives) [reply app/controllers/posts_controller.rb]: i've updated the PR

**Comment by arthurashman** [comment]
I'm happy with what's here. I don't think there's a need to disappear the links at this point

**Pull request merged**